// office-ui-fabric-react/lib/components/ContextualMenu/examples/ContextualMenu.Directional.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Button_1 = require("office-ui-fabric-react/lib/Button");
var Checkbox_1 = require("office-ui-fabric-react/lib/Checkbox");
var ContextualMenu_1 = require("office-ui-fabric-react/lib/ContextualMenu");
var Dropdown_1 = require("office-ui-fabric-react/lib/Dropdown");
var Slider_1 = require("office-ui-fabric-react/lib/Slider");
var Utilities_1 = require("office-ui-fabric-react/lib/Utilities");
require("./ContextualMenuExample.scss");
var DIRECTION_OPTIONS = [
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.topLeftEdge], text: 'Top Left Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.topCenter], text: 'Top Center' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.topRightEdge], text: 'Top Right Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.topAutoEdge], text: 'Top Auto Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.bottomLeftEdge], text: 'Bottom Left Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.bottomCenter], text: 'Bottom Center' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.bottomRightEdge], text: 'Bottom Right Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.bottomAutoEdge], text: 'Bottom Auto Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.leftTopEdge], text: 'Left Top Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.leftCenter], text: 'Left Center' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.leftBottomEdge], text: 'Left Bottom Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.rightTopEdge], text: 'Right Top Edge' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.rightCenter], text: 'Right Center' },
    { key: ContextualMenu_1.DirectionalHint[ContextualMenu_1.DirectionalHint.rightBottomEdge], text: 'Right Bottom Edge' },
];
var ContextualMenuDirectionalExample = (function (_super) {
    __extends(ContextualMenuDirectionalExample, _super);
    function ContextualMenuDirectionalExample() {
        var _this = _super.call(this) || this;
        _this.state = {
            isContextualMenuVisible: false,
            isBeakVisible: false,
            directionalHint: ContextualMenu_1.DirectionalHint.bottomLeftEdge,
            gapSpace: 0,
            beakWidth: 10,
            edgeFixed: false
        };
        return _this;
    }
    ContextualMenuDirectionalExample.prototype.render = function () {
        var _a = this.state, isContextualMenuVisible = _a.isContextualMenuVisible, isBeakVisible = _a.isBeakVisible, directionalHint = _a.directionalHint, gapSpace = _a.gapSpace, beakWidth = _a.beakWidth, edgeFixed = _a.edgeFixed;
        return (React.createElement("div", { className: 'ms-ContextualMenuDirectionalExample' },
            React.createElement("div", { className: 'ms-ContextualMenuDirectionalExample-configArea' },
                React.createElement(Checkbox_1.Checkbox, { label: 'Show beak', checked: isBeakVisible, onChange: this._onShowBeakChange }),
                React.createElement(Checkbox_1.Checkbox, { label: 'Fix Edge', checked: edgeFixed, onChange: this._onFixEdgeChange }),
                React.createElement(Slider_1.Slider, { max: 20, label: 'Gap Space', min: 0, defaultValue: 0, onChange: this._onGapSlider }),
                isBeakVisible &&
                    (React.createElement(Slider_1.Slider, { max: 50, label: 'Beak Width', min: 10, defaultValue: 10, onChange: this._onBeakWidthSlider })),
                React.createElement(Dropdown_1.Dropdown, { label: 'Directional hint', selectedKey: ContextualMenu_1.DirectionalHint[directionalHint], options: DIRECTION_OPTIONS, onChanged: this._onDirectionalChanged })),
            React.createElement("div", { className: 'ms-ContextualMenuDirectionalExample-buttonArea', ref: 'menuButton' },
                React.createElement(Button_1.DefaultButton, { onClick: this._onShowMenuClicked, text: isContextualMenuVisible ? 'Hide context menu' : 'Show context menu' })),
            isContextualMenuVisible ? (React.createElement(ContextualMenu_1.ContextualMenu, { target: this.refs.menuButton, isBeakVisible: isBeakVisible, directionalHint: directionalHint, gapSpace: gapSpace, beakWidth: beakWidth, directionalHintFixed: edgeFixed, items: [
                    {
                        key: 'newItem',
                        name: 'New',
                        icon: 'Add',
                        subMenuProps: {
                            items: [
                                {
                                    key: 'emailMessage',
                                    name: 'Email message',
                                },
                                {
                                    key: 'calendarEvent',
                                    name: 'Calendar event',
                                }
                            ]
                        },
                    },
                    {
                        key: 'upload',
                        name: 'Upload',
                        icon: 'Upload'
                    },
                    {
                        key: 'rename',
                        name: 'Rename',
                    },
                    {
                        key: '-',
                        name: '-',
                    },
                    {
                        key: 'share',
                        name: 'Share',
                        icon: 'Share',
                        items: [
                            {
                                key: 'sharetoemail',
                                name: 'Share to Email',
                                icon: 'Mail'
                            },
                            {
                                key: 'sharetofacebook',
                                name: 'Share to Facebook',
                            },
                            {
                                key: 'sharetotwitter',
                                name: 'Share to Twitter',
                                icon: 'Share'
                            },
                        ]
                    },
                    {
                        key: 'print',
                        name: 'Print',
                        icon: 'Print'
                    },
                    {
                        key: 'display',
                        name: 'display'
                    },
                    {
                        key: 'properties',
                        name: 'Properties'
                    },
                ] })) : (null)));
    };
    ContextualMenuDirectionalExample.prototype._onShowBeakChange = function (ev, isVisible) {
        this.setState({
            isBeakVisible: isVisible
        });
    };
    ContextualMenuDirectionalExample.prototype._onFixEdgeChange = function (ev, isVisible) {
        this.setState({
            edgeFixed: isVisible
        });
    };
    ContextualMenuDirectionalExample.prototype._onShowMenuClicked = function () {
        this.setState({
            isContextualMenuVisible: !this.state.isContextualMenuVisible
        });
    };
    ContextualMenuDirectionalExample.prototype._onDirectionalChanged = function (option) {
        this.setState({
            directionalHint: ContextualMenu_1.DirectionalHint[option.key]
        });
    };
    ContextualMenuDirectionalExample.prototype._onGapSlider = function (value) {
        this.setState({
            gapSpace: value
        });
    };
    ContextualMenuDirectionalExample.prototype._onBeakWidthSlider = function (value) {
        this.setState({
            beakWidth: value
        });
    };
    return ContextualMenuDirectionalExample;
}(React.Component));
__decorate([
    Utilities_1.autobind
], ContextualMenuDirectionalExample.prototype, "_onShowBeakChange", null);
__decorate([
    Utilities_1.autobind
], ContextualMenuDirectionalExample.prototype, "_onFixEdgeChange", null);
__decorate([
    Utilities_1.autobind
], ContextualMenuDirectionalExample.prototype, "_onShowMenuClicked", null);
__decorate([
    Utilities_1.autobind
], ContextualMenuDirectionalExample.prototype, "_onDirectionalChanged", null);
__decorate([
    Utilities_1.autobind
], ContextualMenuDirectionalExample.prototype, "_onGapSlider", null);
__decorate([
    Utilities_1.autobind
], ContextualMenuDirectionalExample.prototype, "_onBeakWidthSlider", null);
exports.ContextualMenuDirectionalExample = ContextualMenuDirectionalExample;
