// office-ui-fabric-react/lib-amd/components/Layer/examples/Layer.Basic.Example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "react", "office-ui-fabric-react/lib/Utilities", "office-ui-fabric-react/lib/Checkbox", "office-ui-fabric-react/lib/Layer", "./Layer.Example.scss"], function (require, exports, React, Utilities_1, Checkbox_1, Layer_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var LayerContentExample = (function (_super) {
        __extends(LayerContentExample, _super);
        function LayerContentExample() {
            var _this = _super.call(this) || this;
            _this.state = {
                time: new Date().toLocaleTimeString()
            };
            return _this;
        }
        LayerContentExample.prototype.componentDidMount = function () {
            var _this = this;
            this._async.setInterval(function () { return _this.setState({ time: new Date().toLocaleTimeString() }); }, 1000);
        };
        LayerContentExample.prototype.render = function () {
            return (React.createElement("div", { className: 'LayerExample-content ms-u-scaleUpIn100' },
                React.createElement("div", { className: 'LayerExample-textContent' }, this.context.message),
                React.createElement("div", null, this.state.time)));
        };
        return LayerContentExample;
    }(Utilities_1.BaseComponent));
    LayerContentExample.contextTypes = {
        message: React.PropTypes.string
    };
    exports.LayerContentExample = LayerContentExample;
    var LayerBasicExample = (function (_super) {
        __extends(LayerBasicExample, _super);
        function LayerBasicExample() {
            var _this = _super.call(this) || this;
            _this.state = {
                showLayer: false
            };
            return _this;
        }
        LayerBasicExample.prototype.getChildContext = function () {
            return {
                'message': 'Hello world.'
            };
        };
        LayerBasicExample.prototype.render = function () {
            var _this = this;
            var showLayer = this.state.showLayer;
            return (React.createElement("div", null,
                React.createElement(Checkbox_1.Checkbox, { label: 'Wrap the content box belowed in a Layer', checked: showLayer, onChange: function (ev, checked) { return _this.setState({ showLayer: checked }); } }),
                showLayer ? (React.createElement(Layer_1.Layer, null,
                    React.createElement(LayerContentExample, null))) : (React.createElement(LayerContentExample, null))));
        };
        return LayerBasicExample;
    }(Utilities_1.BaseComponent));
    LayerBasicExample.childContextTypes = {
        message: React.PropTypes.string
    };
    exports.LayerBasicExample = LayerBasicExample;
});
