// office-ui-fabric-react/lib/components/Facepile/examples/Facepile.Overflow.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Facepile_1 = require("office-ui-fabric-react/lib/Facepile");
var Dropdown_1 = require("office-ui-fabric-react/lib/Dropdown");
var Slider_1 = require("office-ui-fabric-react/lib/Slider");
var FacepileExampleData_1 = require("./FacepileExampleData");
require("./Facepile.Examples.scss");
var facepileProps = {
    personas: FacepileExampleData_1.facepilePersonas,
    maxDisplayablePersonas: 5,
    overflowButtonType: Facepile_1.OverflowButtonType.downArrow,
    overflowButtonProps: {
        ariaLabel: 'More users',
        onClick: function (ev) {
            return alert('overflow icon clicked');
        }
    },
    ariaDescription: 'To move through the items use left and right arrow keys.'
};
var FacepileOverflowExample = (function (_super) {
    __extends(FacepileOverflowExample, _super);
    function FacepileOverflowExample() {
        var _this = _super.call(this) || this;
        _this.state = {
            displayedPersonas: 5,
            overflowButtonType: Facepile_1.OverflowButtonType.none
        };
        return _this;
    }
    FacepileOverflowExample.prototype.render = function () {
        var _this = this;
        var _a = this.state, displayedPersonas = _a.displayedPersonas, overflowButtonType = _a.overflowButtonType;
        facepileProps.maxDisplayablePersonas = displayedPersonas;
        facepileProps.overflowButtonType = overflowButtonType;
        return (React.createElement("div", { className: 'ms-FacepileExample' },
            React.createElement(Facepile_1.Facepile, __assign({}, facepileProps)),
            React.createElement("div", { className: 'control' },
                React.createElement(Slider_1.Slider, { label: 'Number of Personas Shown:', min: 0, max: 6, step: 1, showValue: true, value: this.state.displayedPersonas, onChange: function (value) { return _this.setState(function (prevState) {
                        prevState.displayedPersonas = value;
                        return prevState;
                    }); } })),
            React.createElement(Dropdown_1.Dropdown, { label: 'Overflow Type:', selectedKey: this.state.overflowButtonType, options: [
                    { key: Facepile_1.OverflowButtonType.none, text: Facepile_1.OverflowButtonType[Facepile_1.OverflowButtonType.none] },
                    { key: Facepile_1.OverflowButtonType.descriptive, text: Facepile_1.OverflowButtonType[Facepile_1.OverflowButtonType.descriptive] },
                    { key: Facepile_1.OverflowButtonType.downArrow, text: Facepile_1.OverflowButtonType[Facepile_1.OverflowButtonType.downArrow] },
                    { key: Facepile_1.OverflowButtonType.more, text: Facepile_1.OverflowButtonType[Facepile_1.OverflowButtonType.more] },
                ], onChanged: function (value) { return _this.setState(function (prevState) {
                    prevState.overflowButtonType = value.key;
                    return prevState;
                }); } })));
    };
    return FacepileOverflowExample;
}(React.Component));
exports.FacepileOverflowExample = FacepileOverflowExample;
