// office-ui-fabric-react/lib/components/Persona/examples/Persona.CustomRender.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Utilities_1 = require("office-ui-fabric-react/lib/Utilities");
var Persona_1 = require("office-ui-fabric-react/lib/Persona");
var Icon_1 = require("office-ui-fabric-react/lib/Icon");
require("./PersonaExample.scss");
var examplePersona = {
    imageUrl: './images/persona-female.png',
    imageInitials: 'AL',
    primaryText: 'Annie Lindqvist',
    secondaryText: 'Software Engineer',
    tertiaryText: 'In a meeting',
    optionalText: 'Available at 4:00pm'
};
var PersonaCustomRenderExample = (function (_super) {
    __extends(PersonaCustomRenderExample, _super);
    function PersonaCustomRenderExample() {
        return _super.call(this) || this;
    }
    PersonaCustomRenderExample.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.large, presence: Persona_1.PersonaPresence.offline, onRenderSecondaryText: this._onRenderSecondaryText }))));
    };
    PersonaCustomRenderExample.prototype._onRenderSecondaryText = function (props) {
        return (React.createElement("div", null,
            React.createElement(Icon_1.Icon, { iconName: 'Suitcase', className: 'ms-JobIconExample' }),
            props.secondaryText));
    };
    return PersonaCustomRenderExample;
}(React.Component));
__decorate([
    Utilities_1.autobind
], PersonaCustomRenderExample.prototype, "_onRenderSecondaryText", null);
exports.PersonaCustomRenderExample = PersonaCustomRenderExample;
