// office-ui-fabric-react/lib-amd/components/ResizeGroup/examples/data.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.data = {
        primary: [
            {
                key: 'newItem',
                name: 'New',
                icon: 'Add',
                ariaLabel: 'New. Use left and right arrow keys to navigate',
                onClick: function () { return; },
            },
            {
                key: 'upload',
                name: 'Upload',
                icon: 'Upload',
                onClick: function () { return; },
            },
            {
                key: 'share',
                name: 'Share',
                icon: 'Share',
                onClick: function () { return; }
            },
            {
                key: 'newItem2',
                name: 'New',
                icon: 'Add',
                ariaLabel: 'New. Use left and right arrow keys to navigate',
                onClick: function () { return; },
            },
            {
                key: 'upload2',
                name: 'Upload',
                icon: 'Upload',
                onClick: function () { return; },
            },
            {
                key: 'share2',
                name: 'Share',
                icon: 'Share',
                onClick: function () { return; }
            },
            {
                key: 'newItem3',
                name: 'New',
                icon: 'Add',
                ariaLabel: 'New. Use left and right arrow keys to navigate',
                onClick: function () { return; },
            },
            {
                key: 'upload3',
                name: 'Upload',
                icon: 'Upload',
                onClick: function () { return; },
            },
            {
                key: 'share3',
                name: 'Share',
                icon: 'Share',
                onClick: function () { return; }
            },
            {
                key: 'newItem4',
                name: 'New',
                icon: 'Add',
                ariaLabel: 'New. Use left and right arrow keys to navigate',
                onClick: function () { return; },
            },
            {
                key: 'upload4',
                name: 'Upload',
                icon: 'Upload',
                onClick: function () { return; },
            },
            {
                key: 'share4',
                name: 'Share',
                icon: 'Share',
                onClick: function () { return; }
            }
        ],
        overflow: []
    };
});
