// office-ui-fabric-react/lib/components/Pivot/examples/Pivot.Separate.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Pivot_1 = require("office-ui-fabric-react/lib/Pivot");
var Utilities_1 = require("office-ui-fabric-react/lib/Utilities");
var PivotSeparateExample = (function (_super) {
    __extends(PivotSeparateExample, _super);
    function PivotSeparateExample() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = { selectedKey: 'rectangleRed' };
        return _this;
    }
    PivotSeparateExample.prototype.render = function () {
        return (React.createElement("div", null,
            React.createElement("div", { "aria-labelledby": this._getTabId(this.state.selectedKey), role: 'tabitem', style: {
                    float: 'left',
                    width: 100,
                    height: this.state.selectedKey === 'squareRed' ? 100 : 200,
                    background: this.state.selectedKey === 'rectangleGreen' ? 'green' : 'red'
                } }),
            React.createElement(Pivot_1.Pivot, { selectedKey: this.state.selectedKey, onLinkClick: this._handleLinkClick, headersOnly: true, getTabId: this._getTabId },
                React.createElement(Pivot_1.PivotItem, { linkText: 'Rectangle red', itemKey: 'rectangleRed' }),
                React.createElement(Pivot_1.PivotItem, { linkText: 'Square red', itemKey: 'squareRed' }),
                React.createElement(Pivot_1.PivotItem, { linkText: 'Rectangle green', itemKey: 'rectangleGreen' }))));
    };
    PivotSeparateExample.prototype._handleLinkClick = function (item) {
        this.setState({
            selectedKey: item.props.itemKey
        });
    };
    PivotSeparateExample.prototype._getTabId = function (itemKey) {
        return "ShapeColorPivot_" + itemKey;
    };
    return PivotSeparateExample;
}(React.Component));
__decorate([
    Utilities_1.autobind
], PivotSeparateExample.prototype, "_handleLinkClick", null);
__decorate([
    Utilities_1.autobind
], PivotSeparateExample.prototype, "_getTabId", null);
exports.PivotSeparateExample = PivotSeparateExample;
