// office-ui-fabric-react/lib-amd/components/Persona/examples/Persona.Basic.Example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
define(["require", "exports", "react", "office-ui-fabric-react/lib/Persona", "office-ui-fabric-react/lib/Checkbox"], function (require, exports, React, Persona_1, Checkbox_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var examplePersona = {
        imageUrl: './images/persona-female.png',
        imageInitials: 'AL',
        primaryText: 'Annie Lindqvist',
        secondaryText: 'Software Engineer',
        tertiaryText: 'In a meeting',
        optionalText: 'Available at 4:00pm'
    };
    var PersonaBasicExample = (function (_super) {
        __extends(PersonaBasicExample, _super);
        function PersonaBasicExample() {
            var _this = _super.call(this) || this;
            _this.state = {
                renderPersonaDetails: true
            };
            return _this;
        }
        PersonaBasicExample.prototype.render = function () {
            var _this = this;
            var renderPersonaDetails = this.state.renderPersonaDetails;
            return (React.createElement("div", null,
                React.createElement("div", null,
                    React.createElement(Checkbox_1.Checkbox, { label: 'Include persona details', checked: renderPersonaDetails, onChange: function (ev, checked) { _this.setState({ renderPersonaDetails: checked }); } })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.tiny, presence: Persona_1.PersonaPresence.offline, hidePersonaDetails: !renderPersonaDetails })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.extraExtraSmall, presence: Persona_1.PersonaPresence.none, hidePersonaDetails: !renderPersonaDetails })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.size28, presence: Persona_1.PersonaPresence.none, hidePersonaDetails: !renderPersonaDetails })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.extraSmall, presence: Persona_1.PersonaPresence.online, hidePersonaDetails: !renderPersonaDetails })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.small, presence: Persona_1.PersonaPresence.away, hidePersonaDetails: !renderPersonaDetails })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { hidePersonaDetails: !renderPersonaDetails, presence: Persona_1.PersonaPresence.busy })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.large, presence: Persona_1.PersonaPresence.dnd, hidePersonaDetails: !renderPersonaDetails })),
                React.createElement(Persona_1.Persona, __assign({}, examplePersona, { size: Persona_1.PersonaSize.extraLarge, presence: Persona_1.PersonaPresence.blocked, hidePersonaDetails: !renderPersonaDetails }))));
        };
        return PersonaBasicExample;
    }(React.Component));
    exports.PersonaBasicExample = PersonaBasicExample;
});
