// office-ui-fabric-react/lib-amd/components/DocumentCard/DocumentCardPage.visualtest.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
define(["require", "exports", "./index", "react"], function (require, exports, index_1, React) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    /* tslint:enable:no-unused-variable */
    var DocumentCardVPage = (function (_super) {
        __extends(DocumentCardVPage, _super);
        function DocumentCardVPage() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        DocumentCardVPage.prototype.render = function () {
            var previewProps = {
                previewImages: [
                    {
                        name: 'DocumentCard',
                        previewImageSrc: 'dist/document-preview.png',
                        iconSrc: 'dist/icon-ppt.png',
                        width: 318,
                        height: 196,
                        accentColor: '#ce4b1f'
                    }
                ],
            };
            return React.createElement("div", null,
                React.createElement(index_1.DocumentCard, { className: 'DocumentCard' },
                    React.createElement(index_1.DocumentCardPreview, __assign({}, previewProps)),
                    React.createElement(index_1.DocumentCardTitle, { title: 'Document Card', shouldTruncate: true }),
                    React.createElement(index_1.DocumentCardActivity, { activity: 'Created a few minutes ago', people: [
                            { name: 'Annie Lindqvist', profileImageSrc: 'images/persona-female.png' }
                        ] })));
        };
        return DocumentCardVPage;
    }(React.Component));
    exports.default = DocumentCardVPage;
});
