// office-ui-fabric-react/lib-amd/components/ChoiceGroup/examples/ChoiceGroup.Image.Example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "react", "office-ui-fabric-react/lib/ChoiceGroup"], function (require, exports, React, ChoiceGroup_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ChoiceGroupImageExample = (function (_super) {
        __extends(ChoiceGroupImageExample, _super);
        function ChoiceGroupImageExample() {
            var _this = _super.call(this) || this;
            _this.state = {
                selectedKey: 'bar'
            };
            _this._onImageChoiceGroupChange = _this._onImageChoiceGroupChange.bind(_this);
            return _this;
        }
        ChoiceGroupImageExample.prototype.render = function () {
            var selectedKey = this.state.selectedKey;
            return (React.createElement("div", null,
                React.createElement(ChoiceGroup_1.ChoiceGroup, { label: 'Pick one image', selectedKey: selectedKey, options: [
                        {
                            key: 'bar',
                            imageSrc: './images/choicegroup-bar-unselected.png',
                            selectedImageSrc: './images/choicegroup-bar-selected.png',
                            imageSize: { width: 32, height: 32 },
                            text: 'Bar chart'
                        },
                        {
                            key: 'pie',
                            imageSrc: './images/choicegroup-pie-unselected.png',
                            selectedImageSrc: './images/choicegroup-pie-selected.png',
                            imageSize: { width: 32, height: 32 },
                            text: 'Pie chart'
                        }
                    ], onChange: this._onImageChoiceGroupChange })));
        };
        ChoiceGroupImageExample.prototype._onImageChoiceGroupChange = function (ev, option) {
            this.setState({
                selectedKey: option.key
            });
        };
        return ChoiceGroupImageExample;
    }(React.Component));
    exports.ChoiceGroupImageExample = ChoiceGroupImageExample;
});
