// office-ui-fabric-react/lib-amd/components/ChoiceGroup/examples/ChoiceGroup.Custom.Example.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
define(["require", "exports", "react", "office-ui-fabric-react/lib/ChoiceGroup", "office-ui-fabric-react/lib/Utilities", "office-ui-fabric-react/lib/Dropdown", "./ChoiceGroup.Custom.Example.scss"], function (require, exports, React, ChoiceGroup_1, Utilities_1, Dropdown_1, stylesImport) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var styles = stylesImport;
    var ChoiceGroupCustomExample = (function (_super) {
        __extends(ChoiceGroupCustomExample, _super);
        function ChoiceGroupCustomExample() {
            return _super.call(this) || this;
        }
        ChoiceGroupCustomExample.prototype.render = function () {
            return (React.createElement("div", null,
                React.createElement(ChoiceGroup_1.ChoiceGroup, { defaultSelectedKey: 'B', options: [
                        {
                            key: 'A',
                            text: 'Mark displayed items as read after',
                            onRenderField: function (props, render) {
                                return (React.createElement("div", { className: Utilities_1.css(styles.root) },
                                    render(props),
                                    React.createElement(Dropdown_1.Dropdown, { className: Utilities_1.css(styles.dropdown), defaultSelectedKey: 'A', options: [
                                            { key: 'A', text: '5 seconds' },
                                            { key: 'B', text: '10 seconds' },
                                            { key: 'C', text: '20 seconds' }
                                        ], disabled: false })));
                            }
                        },
                        {
                            key: 'B',
                            text: 'Option B',
                        },
                        {
                            key: 'C',
                            text: 'Option C',
                            disabled: true
                        },
                        {
                            key: 'D',
                            text: 'Option D',
                            disabled: false
                        }
                    ], onChange: this._onChange, label: 'Pick one', required: true })));
        };
        ChoiceGroupCustomExample.prototype._onChange = function (ev, option) {
            console.dir(option);
        };
        return ChoiceGroupCustomExample;
    }(React.Component));
    __decorate([
        Utilities_1.autobind
    ], ChoiceGroupCustomExample.prototype, "_onChange", null);
    exports.ChoiceGroupCustomExample = ChoiceGroupCustomExample;
});
