// office-ui-fabric-react/lib/components/Panel/examples/Panel.SmallRight.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Button_1 = require("office-ui-fabric-react/lib/Button");
var ChoiceGroup_1 = require("office-ui-fabric-react/lib/ChoiceGroup");
var Panel_1 = require("office-ui-fabric-react/lib/Panel");
var PanelSmallRightExample = (function (_super) {
    __extends(PanelSmallRightExample, _super);
    function PanelSmallRightExample() {
        var _this = _super.call(this) || this;
        _this._onClosePanel = function () {
            _this.setState({ showPanel: false });
        };
        _this._onShowPanel = function () {
            _this.setState({ showPanel: true });
        };
        _this.state = {
            showPanel: false
        };
        return _this;
    }
    PanelSmallRightExample.prototype.render = function () {
        var _this = this;
        return (React.createElement("div", null,
            React.createElement(Button_1.DefaultButton, { description: 'Opens the Sample Panel', onClick: this._onShowPanel, text: 'Open Panel' }),
            React.createElement(Panel_1.Panel, { isOpen: this.state.showPanel, type: Panel_1.PanelType.smallFixedFar, onDismiss: this._onClosePanel, headerText: 'Panel - Small, right-aligned, fixed, with footer', closeButtonAriaLabel: 'Close', onRenderFooterContent: function () {
                    return (React.createElement("div", null,
                        React.createElement(Button_1.PrimaryButton, { onClick: _this._onClosePanel, style: { 'marginRight': '8px' } }, "Save"),
                        React.createElement(Button_1.DefaultButton, { onClick: _this._onClosePanel }, "Cancel")));
                } },
                React.createElement(ChoiceGroup_1.ChoiceGroup, { options: [
                        {
                            key: 'A',
                            text: 'Option A'
                        },
                        {
                            key: 'B',
                            text: 'Option B',
                            checked: true
                        },
                        {
                            key: 'C',
                            text: 'Option C',
                            disabled: true
                        },
                        {
                            key: 'D',
                            text: 'Option D',
                            checked: true,
                            disabled: true
                        }
                    ], label: 'Pick one', required: true }))));
    };
    return PanelSmallRightExample;
}(React.Component));
exports.PanelSmallRightExample = PanelSmallRightExample;
