// office-ui-fabric-react/lib-amd/demo/AppDefinition.js
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
define(["require", "exports", "react", "@uifabric/example-app-base", "../components/DetailsList/examples/DetailsList.Basic.Example"], function (require, exports, React, example_app_base_1, DetailsList_Basic_Example_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.AppDefinition = {
        appTitle: 'Fabric - React',
        testPages: [
            {
                component: DetailsList_Basic_Example_1.DetailsListBasicExample,
                key: 'DetailsListBasicExample',
                name: 'DetailsListBasicExample',
                url: '#/tests/detailslistbasicexample'
            }
        ],
        examplePages: [
            {
                links: [
                    {
                        component: require('../components/Breadcrumb/BreadcrumbPage').BreadcrumbPage,
                        key: 'Breadcrumb',
                        name: 'Breadcrumb',
                        url: '#/examples/breadcrumb'
                    },
                    {
                        component: require('../components/Button/ButtonPage').ButtonPage,
                        key: 'Button',
                        name: 'Button',
                        url: '#/examples/button'
                    },
                    {
                        component: require('../components/Calendar/CalendarPage').CalendarPage,
                        key: 'Calendar',
                        name: 'Calendar',
                        url: '#/examples/calendar'
                    },
                    {
                        component: require('../components/Callout/CalloutPage').CalloutPage,
                        key: 'Callout',
                        name: 'Callout',
                        url: '#/examples/callout'
                    },
                    {
                        component: require('../components/Checkbox/CheckboxPage').CheckboxPage,
                        key: 'Checkbox',
                        name: 'Checkbox',
                        url: '#/examples/checkbox'
                    },
                    {
                        component: require('../components/ChoiceGroup/ChoiceGroupPage').ChoiceGroupPage,
                        key: 'ChoiceGroup',
                        name: 'ChoiceGroup',
                        url: '#/examples/choicegroup'
                    },
                    {
                        component: require('../components/CommandBar/CommandBarPage').CommandBarPage,
                        key: 'CommandBar',
                        name: 'CommandBar',
                        url: '#/examples/commandbar'
                    },
                    {
                        component: require('../components/ContextualMenu/ContextualMenuPage').ContextualMenuPage,
                        key: 'ContextualMenu',
                        name: 'ContextualMenu',
                        url: '#/examples/contextmenu'
                    },
                    {
                        component: require('../components/DatePicker/DatePickerPage').DatePickerPage,
                        key: 'DatePicker',
                        name: 'DatePicker',
                        url: '#/examples/datepicker'
                    },
                    {
                        component: require('../components/DetailsList/DetailsListPage').DetailsListPage,
                        key: 'DetailsList',
                        name: 'DetailsList',
                        url: '#/examples/detailslist'
                    },
                    {
                        component: require('../components/Dialog/DialogPage').DialogPage,
                        key: 'Dialog',
                        name: 'Dialog',
                        url: '#/examples/dialog'
                    },
                    {
                        component: require('../components/DocumentCard/DocumentCardPage').DocumentCardPage,
                        key: 'DocumentCard',
                        name: 'DocumentCard',
                        url: '#/examples/documentcard'
                    },
                    {
                        component: require('../components/Dropdown/DropdownPage').DropdownPage,
                        key: 'Dropdown',
                        name: 'Dropdown',
                        url: '#/examples/dropdown'
                    },
                    {
                        component: require('../components/Facepile/FacepilePage').FacepilePage,
                        key: 'Facepile',
                        name: 'Facepile',
                        url: '#/examples/facepile'
                    },
                    {
                        component: require('../components/Icon/IconPage').IconPage,
                        key: 'Icon',
                        name: 'Icon',
                        url: '#/examples/icon'
                    },
                    {
                        component: require('../components/Label/LabelPage').LabelPage,
                        key: 'Label',
                        name: 'Label',
                        url: '#/examples/label'
                    },
                    {
                        component: require('../components/Link/LinkPage').LinkPage,
                        key: 'Link',
                        name: 'Link',
                        url: '#/examples/link'
                    },
                    {
                        component: require('../components/List/ListPage').ListPage,
                        key: 'List',
                        name: 'List',
                        url: '#/examples/list'
                    },
                    {
                        component: require('../components/MessageBar/MessageBarPage').MessageBarPage,
                        key: 'MessageBar',
                        name: 'MessageBar',
                        url: '#/examples/messagebar'
                    },
                    {
                        component: require('../components/Modal/ModalPage').ModalPage,
                        key: 'Modal',
                        name: 'Modal',
                        url: '#/examples/modal'
                    },
                    {
                        component: require('../components/Overlay/OverlayPage').OverlayPage,
                        key: 'Overlay',
                        name: 'Overlay',
                        url: '#/examples/overlay'
                    },
                    {
                        component: require('../components/OverflowSet/OverflowSetPage').OverflowSetPage,
                        key: 'OverflowSet',
                        name: 'OverflowSet',
                        url: '#/examples/overflowset'
                    },
                    {
                        component: require('../components/Panel/PanelPage').PanelPage,
                        key: 'Panel',
                        name: 'Panel',
                        url: '#/examples/panel'
                    },
                    {
                        component: require('../components/pickers/PickersPage').PickersPage,
                        key: 'Pickers',
                        name: 'Pickers',
                        url: '#/examples/pickers'
                    },
                    {
                        component: require('../components/pickers/PeoplePicker/PeoplePickerPage').PeoplePickerPage,
                        key: 'PeoplePicker',
                        name: 'PeoplePicker',
                        url: '#/examples/PeoplePicker'
                    },
                    {
                        component: require('../components/Persona/PersonaPage').PersonaPage,
                        key: 'Persona',
                        name: 'Persona',
                        url: '#/examples/persona'
                    },
                    {
                        component: require('../components/Pivot/PivotPage').PivotPage,
                        key: 'Pivot',
                        name: 'Pivot',
                        url: '#/examples/pivot'
                    },
                    {
                        component: require('../components/ProgressIndicator/ProgressIndicatorPage').ProgressIndicatorPage,
                        key: 'ProgressIndicator',
                        name: 'ProgressIndicator',
                        url: '#/examples/progressindicator'
                    },
                    {
                        component: require('../components/Rating/RatingPage').RatingPage,
                        key: 'Rating',
                        name: 'Rating',
                        url: '#/examples/rating'
                    },
                    {
                        component: require('../components/SearchBox/SearchBoxPage').SearchBoxPage,
                        key: 'SearchBox',
                        name: 'SearchBox',
                        url: '#/examples/searchbox'
                    },
                    {
                        component: require('../components/Spinner/SpinnerPage').SpinnerPage,
                        key: 'Spinner',
                        name: 'Spinner',
                        url: '#/examples/spinner'
                    },
                    {
                        component: require('../components/TeachingBubble/TeachingBubblePage').TeachingBubblePage,
                        key: 'TeachingBubble',
                        name: 'TeachingBubble',
                        url: '#/examples/teachingbubble'
                    },
                    {
                        component: require('../components/TextField/TextFieldPage').TextFieldPage,
                        key: 'TextField',
                        name: 'TextField',
                        url: '#/examples/textfield'
                    },
                    {
                        component: require('../components/Toggle/TogglePage').TogglePage,
                        key: 'Toggle',
                        name: 'Toggle',
                        url: '#/examples/toggle'
                    },
                    {
                        component: require('../components/Tooltip/TooltipPage').TooltipPage,
                        key: 'Tooltip',
                        name: 'Tooltip',
                        url: '#/examples/Tooltip'
                    }
                ],
                name: 'Basic components'
            },
            {
                links: [
                    {
                        component: require('../components/ColorPicker/ColorPickerPage').ColorPickerPage,
                        key: 'ColorPicker',
                        name: 'ColorPicker',
                        url: '#/examples/colorpicker'
                    },
                    {
                        component: require('../components/GroupedList/GroupedListPage').GroupedListPage,
                        key: 'GroupedList',
                        name: 'GroupedList',
                        url: '#examples/groupedlist'
                    },
                    {
                        component: require('../components/Image/ImagePage').ImagePage,
                        key: 'Image',
                        name: 'Image',
                        url: '#/examples/image'
                    },
                    {
                        component: require('../components/Layer/LayerPage').LayerPage,
                        key: 'Layer',
                        name: 'Layer',
                        url: '#/examples/layer'
                    },
                    {
                        component: require('../components/Nav/NavPage').NavPage,
                        key: 'Nav',
                        name: 'Nav',
                        url: '#/examples/nav'
                    },
                    {
                        component: require('../components/Slider/SliderPage').SliderPage,
                        key: 'Slider',
                        name: 'Slider',
                        url: '#/examples/slider'
                    }
                ],
                name: 'Extended components'
            },
            {
                links: [
                    {
                        component: require('../components/FocusTrapZone/FocusTrapZonePage').FocusTrapZonePage,
                        key: 'FocusTrapZone',
                        name: 'FocusTrapZone',
                        url: '#examples/focustrapzone'
                    },
                    {
                        component: require('../components/FocusZone/FocusZonePage').FocusZonePage,
                        key: 'FocusZone',
                        name: 'FocusZone',
                        url: '#examples/focuszone'
                    },
                    {
                        component: require('../components/MarqueeSelection/MarqueeSelectionPage').MarqueeSelectionPage,
                        key: 'MarqueeSelection',
                        name: 'MarqueeSelection',
                        url: '#examples/marqueeselection'
                    },
                    {
                        component: require('../utilities/selection/SelectionPage').SelectionPage,
                        key: 'Selection',
                        name: 'Selection',
                        url: '#examples/selection'
                    },
                    {
                        component: require('../components/Theme/ThemePage').ThemePage,
                        key: 'Theme',
                        name: 'Themes',
                        url: '#examples/themes'
                    }
                ],
                name: 'Utilities'
            }
        ],
        headerLinks: [
            {
                name: 'Getting started',
                url: '#/'
            },
            {
                name: 'Fabric',
                url: 'http://dev.office.com/fabric'
            },
            {
                name: 'Github',
                url: 'http://www.github.com/officedev'
            }
        ]
    };
    exports.App = function (props) { return React.createElement(example_app_base_1.App, __assign({ appDefinition: exports.AppDefinition }, props)); };
});
