// office-ui-fabric-react/lib/components/Image/examples/Image.None.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Image_1 = require("office-ui-fabric-react/lib/Image");
var Label_1 = require("office-ui-fabric-react/lib/Label");
var ImageNoneExample = (function (_super) {
    __extends(ImageNoneExample, _super);
    function ImageNoneExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ImageNoneExample.prototype.render = function () {
        var imageProps = {
            src: 'http://placehold.it/500x250',
            imageFit: Image_1.ImageFit.none,
            width: 350,
            height: 150
        };
        return (React.createElement("div", null,
            React.createElement("p", null, "By setting the imageFit property to \"none\", the image will remain at its natural size, even if the frame is made larger or smaller by setting the width and height props."),
            React.createElement(Label_1.Label, null, "The image is larger than the frame, so it is cropped to fit. The image is positioned at the upper left of the frame."),
            React.createElement(Image_1.Image, __assign({}, imageProps)),
            React.createElement("br", null),
            React.createElement(Label_1.Label, null, "The image is smaller than the frame, so there is empty space within the frame. The image is positioned at the upper left of the frame."),
            React.createElement(Image_1.Image, __assign({}, imageProps, { src: 'http://placehold.it/100x100' }))));
    };
    return ImageNoneExample;
}(React.Component));
exports.ImageNoneExample = ImageNoneExample;
