// office-ui-fabric-react/lib/components/Image/examples/Image.Center.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Image_1 = require("office-ui-fabric-react/lib/Image");
var Label_1 = require("office-ui-fabric-react/lib/Label");
var ImageCenterExample = (function (_super) {
    __extends(ImageCenterExample, _super);
    function ImageCenterExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ImageCenterExample.prototype.render = function () {
        var imageProps = {
            src: 'http://placehold.it/800x300',
            imageFit: Image_1.ImageFit.center,
            width: 350,
            height: 150,
            onLoad: function (ev) { return console.log('image loaded', ev); }
        };
        return (React.createElement("div", null,
            React.createElement("p", null, "Setting the imageFit property to \"center\" behaves the same as \"none\", while centering the image within the frame."),
            React.createElement(Label_1.Label, null, "The image is larger than the frame, so all sides are cropped to center the image."),
            React.createElement(Image_1.Image, __assign({}, imageProps, { src: 'http://placehold.it/800x300' })),
            React.createElement("br", null),
            React.createElement(Label_1.Label, null, "The image is smaller than the frame, so there is empty space within the frame. The image is centered in the available space."),
            React.createElement(Image_1.Image, __assign({}, imageProps, { src: 'http://placehold.it/100x100' }))));
    };
    return ImageCenterExample;
}(React.Component));
exports.ImageCenterExample = ImageCenterExample;
