// office-ui-fabric-react/lib/components/DatePicker/examples/DatePicker.Required.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var DatePicker_1 = require("office-ui-fabric-react/lib/DatePicker");
var DayPickerStrings = {
    months: [
        'January',
        'February',
        'March',
        'April',
        'May',
        'June',
        'July',
        'August',
        'September',
        'October',
        'November',
        'December'
    ],
    shortMonths: [
        'Jan',
        'Feb',
        'Mar',
        'Apr',
        'May',
        'Jun',
        'Jul',
        'Aug',
        'Sep',
        'Oct',
        'Nov',
        'Dec'
    ],
    days: [
        'Sunday',
        'Monday',
        'Tuesday',
        'Wednesday',
        'Thursday',
        'Friday',
        'Saturday'
    ],
    shortDays: [
        'S',
        'M',
        'T',
        'W',
        'T',
        'F',
        'S'
    ],
    goToToday: 'Go to today',
    prevMonthAriaLabel: 'Go to previous month',
    nextMonthAriaLabel: 'Go to next month',
    prevYearAriaLabel: 'Go to previous year',
    nextYearAriaLabel: 'Go to next year',
    isRequiredErrorMessage: 'Field is required.',
    invalidInputErrorMessage: 'Invalid date format.'
};
var DatePickerRequiredExample = (function (_super) {
    __extends(DatePickerRequiredExample, _super);
    function DatePickerRequiredExample() {
        var _this = _super.call(this) || this;
        _this.state = {
            firstDayOfWeek: DatePicker_1.DayOfWeek.Sunday
        };
        return _this;
    }
    DatePickerRequiredExample.prototype.render = function () {
        var firstDayOfWeek = this.state.firstDayOfWeek;
        return (React.createElement("div", null,
            React.createElement("p", null, "Validation will happen when Date Picker loses focus."),
            React.createElement(DatePicker_1.DatePicker, { label: 'Date required', isRequired: true, firstDayOfWeek: firstDayOfWeek, strings: DayPickerStrings, placeholder: 'Select a date...' })));
    };
    return DatePickerRequiredExample;
}(React.Component));
exports.DatePickerRequiredExample = DatePickerRequiredExample;
