// office-ui-fabric-react/lib/components/CommandBar/CommandBar.visualtest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var VisualTest_1 = require("../../common/VisualTest");
var RunVisualTest_1 = require("../../visualtest/RunVisualTest");
var componentIds = [];
componentIds.push({
    selector: '.' + 'ms-CommandBarItem-link',
    imageSelector: '.' + 'CommandBar',
    fileName: 'commandBar',
    commands: [RunVisualTest_1.defaultScreenshot, RunVisualTest_1.mouseMoveScreenshot, RunVisualTest_1.mouseDownScreenshot, RunVisualTest_1.mouseClickScreenshot]
});
componentIds.push({
    selector: '.' + 'ms-CommandBarItem-link',
    imageSelector: '.' + 'CommandBar',
    fileName: 'commandBarItem',
    commands: [RunVisualTest_1.mouseSingleClickScreenshot],
    childParams: {
        selector: '.' + 'ms-ContextualMenu-list',
        imageSelector: '.' + 'ms-ContextualMenu-Callout',
        fileName: 'commandBarItem',
        commands: [RunVisualTest_1.defaultScreenshot],
    }
});
casper.
    start(VisualTest_1.baseUrl + 'commandBar').
    then(function () {
    RunVisualTest_1.testRunner(componentIds);
});
casper.run(function () { casper.test.done(); });
