// office-ui-fabric-react/lib/components/CommandBar/CommandBarPage.visualtest.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = require("./index");
/* tslint:disable:no-unused-variable */
var React = require("react");
/* tslint:enable:no-unused-variable */
var CommandBarVPage = (function (_super) {
    __extends(CommandBarVPage, _super);
    function CommandBarVPage() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CommandBarVPage.prototype.render = function () {
        return React.createElement("div", { style: { width: '600px' } },
            React.createElement(index_1.CommandBar, { className: 'CommandBar', items: [
                    {
                        key: 'new',
                        name: 'New',
                        icon: 'Add',
                        onClick: function () { return; },
                        items: [
                            {
                                key: 'emailMessage',
                                name: 'Email message',
                                icon: 'Mail'
                            },
                            {
                                key: 'calendarEvent',
                                name: 'Calendar event',
                                icon: 'Calendar'
                            }
                        ]
                    },
                    {
                        key: 'upload',
                        name: 'Upload',
                        icon: 'Upload'
                    }
                ] }));
    };
    return CommandBarVPage;
}(React.Component));
exports.default = CommandBarVPage;
