// office-ui-fabric-react/lib/components/Nav/examples/Nav.ByKeys.Example.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Nav_1 = require("office-ui-fabric-react/lib/Nav");
var NavByKeysExample = (function (_super) {
    __extends(NavByKeysExample, _super);
    function NavByKeysExample() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NavByKeysExample.prototype.render = function () {
        return (React.createElement(Nav_1.Nav, { groups: [{
                    links: [
                        { name: 'Home', key: 'Home', url: '' },
                        { name: 'Activity', key: 'Activity', url: '' },
                        { name: 'News', key: 'News', url: '' },
                        { name: 'Documents', key: 'Documents', url: '' },
                    ]
                }] }));
    };
    return NavByKeysExample;
}(React.Component));
exports.NavByKeysExample = NavByKeysExample;
