// office-ui-fabric-react/lib-amd/components/Modal/Modal.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
define(["require", "exports", "react", "../../Utilities", "../FocusTrapZone/index", "../../Overlay", "../../Layer", "../Popup/index", "../../utilities/decorators/withResponsiveMode", "./Modal.scss"], function (require, exports, React, Utilities_1, index_1, Overlay_1, Layer_1, index_2, withResponsiveMode_1, stylesImport) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var styles = stylesImport;
    var Modal = (function (_super) {
        __extends(Modal, _super);
        function Modal(props) {
            var _this = _super.call(this, props) || this;
            _this.state = {
                id: Utilities_1.getId('Modal'),
                isOpen: props.isOpen,
                isVisible: props.isOpen,
            };
            return _this;
        }
        Modal.prototype.componentWillReceiveProps = function (newProps) {
            clearTimeout(this._onModalCloseTimer);
            // Opening the dialog
            if (newProps.isOpen) {
                if (!this.state.isOpen) {
                    // First Open
                    this.setState({
                        isOpen: true
                    });
                }
                else {
                    // Reopen during closing
                    this.setState({
                        isVisible: true
                    });
                }
            }
            // Closing the dialog
            if (!newProps.isOpen && this.state.isOpen) {
                this._onModalCloseTimer = this._async.setTimeout(this._onModalClose, parseFloat(styles.duration) * 1000);
                this.setState({
                    isVisible: false
                });
            }
        };
        Modal.prototype.componentDidUpdate = function (prevProps, prevState) {
            if (!prevProps.isOpen && !prevState.isVisible) {
                this.setState({
                    isVisible: true
                });
            }
        };
        Modal.prototype.render = function () {
            var _a = this.props, elementToFocusOnDismiss = _a.elementToFocusOnDismiss, firstFocusableSelector = _a.firstFocusableSelector, forceFocusInsideTrap = _a.forceFocusInsideTrap, ignoreExternalFocusing = _a.ignoreExternalFocusing, isBlocking = _a.isBlocking, isClickableOutsideFocusTrap = _a.isClickableOutsideFocusTrap, isDarkOverlay = _a.isDarkOverlay, onDismiss = _a.onDismiss, onLayerDidMount = _a.onLayerDidMount, responsiveMode = _a.responsiveMode, titleAriaId = _a.titleAriaId, subtitleAriaId = _a.subtitleAriaId;
            var _b = this.state, id = _b.id, isOpen = _b.isOpen, isVisible = _b.isVisible;
            var modalClassName = Utilities_1.css('ms-Dialog', styles.root, this.props.className, (_c = {},
                _c['is-open'] = isOpen,
                _c[styles.rootIsVisible] = isVisible,
                _c));
            if (!isOpen) {
                return null;
            }
            // @temp tuatology - Will adjust this to be a panel at certain breakpoints
            if (responsiveMode >= withResponsiveMode_1.ResponsiveMode.small) {
                return (React.createElement(Layer_1.Layer, { onLayerDidMount: onLayerDidMount },
                    React.createElement(index_2.Popup, { role: isBlocking ? 'alertdialog' : 'dialog', ariaLabelledBy: titleAriaId, ariaDescribedBy: subtitleAriaId, onDismiss: onDismiss },
                        React.createElement("div", { className: modalClassName },
                            React.createElement(Overlay_1.Overlay, { isDarkThemed: isDarkOverlay, onClick: isBlocking ? null : onDismiss }),
                            React.createElement(index_1.FocusTrapZone, { className: Utilities_1.css('ms-Dialog-main', styles.main, this.props.containerClassName), elementToFocusOnDismiss: elementToFocusOnDismiss, isClickableOutsideFocusTrap: isClickableOutsideFocusTrap ? isClickableOutsideFocusTrap : !isBlocking, ignoreExternalFocusing: ignoreExternalFocusing, forceFocusInsideTrap: forceFocusInsideTrap, firstFocusableSelector: firstFocusableSelector }, this.props.children)))));
            }
            var _c;
        };
        // Watch for completed animations and set the state
        Modal.prototype._onModalClose = function () {
            this.setState({
                isOpen: false
            });
            // Call the onDismiss callback
            if (this.props.onDismissed) {
                this.props.onDismissed();
            }
        };
        return Modal;
    }(Utilities_1.BaseComponent));
    Modal.defaultProps = {
        isOpen: false,
        isDarkOverlay: true,
        isBlocking: false,
        className: '',
        containerClassName: '',
    };
    Modal = __decorate([
        withResponsiveMode_1.withResponsiveMode
    ], Modal);
    exports.Modal = Modal;
});
